src/MLOps.NET/Storage/Repositories/DataRepository.cs

```diff
@@ -71,5 +73,67 @@ public Data GetData(Guid runId)
 
             return data;
         }
+
+        ///<inheritdoc cref="IDataRepository"/>
+        public async Task LogDataDistribution<T>(Guid runId, IDataView dataView, string columnName) where T : struct
+        {
+            using var db = this.contextFactory.CreateDbContext();
+
+            var data = db.Data
```

Comment:
Let's add two checks here
1) The `Data` exist for a run (e.g. they have logged it already) and if not return an exception with a good message
2) The `DataColumn` exist for the data, and if not, return an exception again